File: leetcode/UnionFind/leetcode-0128/128.go
package unionfind

func longestConsecutive(nums []int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}

	set := make(map[int]int)
	for _, num := range nums {
		set[num] = 1
	}

	res := 0
	for _, num := range nums {
		if _, ok := set[num]; !ok {
			continue
		}
		delete(set, num)
		pre := num - 1
		next := num + 1
		for set[pre] > 0 {
			delete(set, pre)
			pre--
		}
		for set[next] > 0 {
			delete(set, next)
			next++
		}

		if next-pre-1 > res {
			res = next - pre - 1
		}
	}
	return res
}

File: leetcode/UnionFind/leetcode-0128/128.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }

        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int res = 0;
        for (int num : nums) {
            if (!set.contains(num)) {
                continue;
            }
            set.remove(num);
            int pre = num - 1;
            int next = num + 1;
            while (set.contains(pre)) {
                set.remove(pre);
                pre--;
            }
            while (set.contains(next)) {
                set.remove(next);
                next++;
            }
            res = Math.max(res, next - pre - 1);
        }
        return res;
    }
}